bootstrap/api/swagger.yaml

```diff
@@ -0,0 +1,133 @@
+swagger: "2.0"
+info:
+  description: "Deployment API."
+  version: "1.0.0"
+  title: "Kubeflow Deployment"
+  termsOfService: "https://www.kubeflow.org/"
+  contact:
+    email: "kubeflow-engineering@google.com"
+  license:
+    name: "Apache 2.0"
+    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+host: "https://www.kubeflow.org/"
+basePath: "/kfctl"
+tags:
+- name: "deployment"
+  description: "A kubeflow deployment on GCP"
+  externalDocs:
+    description: "Find out more"
+    url: "https://www.kubeflow.org/docs/started/getting-started-gke/"
+schemes:
+- "https"
+paths:
+  /e2eDeploy:
+    post:
+      tags:
+      - "deployment"
+      summary: "Create a new kubeflow deployment on GCP"
+      description: ""
+      operationId: "createDeployment"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: "body"
+        name: "body"
+        description: "Pet object that needs to be added to the store"
+        required: true
+        schema:
+          $ref: "#/definitions/CreateDeployment"
+      responses:
+        200:
+          description: "successful operation"
```

Comment:
One of the goals of the deployment process is to split it into separate phases generate and apply so that people can manually edit the manifests after generate in order to customize them.

Any thought how we could handle that?

One option would be to create custom verbs e.g generate & apply. But that goes against the K8s idea of having a common resource model with common verbs.

A second option would be to have a field in the spec that specifies the desiredState for the deployment. Allowed values could be something like "configured" "deployed"

The flow could then be something like
1. Call create with desiredState: configured
2. This creates the manifests and pushes them to source repo
3. User modifies the manifests
4. User calls Update setting the desiredState to deployed